script.js

```diff
@@ -0,0 +1,167 @@
+const countriesGrid = document.querySelector('.countries-grid');
+let countryFetchedData = null;
+
+async function fetchCountriesData() {
+    if (countryFetchedData) {
+        return;
+    }
+    try {
```

Comment:
1) you can write this like  `if (countryFetchedData) return` 

2) This function should do 1 thing and that fetch the data I recommend writing your logic before calling the function